File: Chapter11/R/Example1.R
# Chapter 11 - Σημειακή Εκτίμηση
# Example 1: Βασικοί Εκτιμητές
#   N(100, 15^2), n=50: x̄, S², σ̂²_MLE, SE(x̄)

# Simulated sample from N(100, 15^2)
set.seed(42)
n           <- 50
sample_data <- rnorm(n, mean = 100, sd = 15)

# Point estimates for mean
mean_estimate <- mean(sample_data)
cat(sprintf("Mean estimate: %.4f\n", mean_estimate))

# Point estimates for variance
# Unbiased estimator S^2 (divides by n-1)
var_unbiased <- var(sample_data)
# MLE estimator (divides by n)
var_mle <- var(sample_data) * (n - 1) / n

cat(sprintf("Unbiased variance (S^2): %.4f\n", var_unbiased))
cat(sprintf("MLE variance:            %.4f\n", var_mle))

# Standard deviation estimates
sd_unbiased <- sd(sample_data)
sd_mle      <- sqrt(var_mle)

cat(sprintf("Unbiased SD: %.4f\n", sd_unbiased))
cat(sprintf("MLE SD:      %.4f\n", sd_mle))

# Standard error of the mean
se_mean <- sd(sample_data) / sqrt(n)
cat(sprintf("SE of mean:  %.4f\n", se_mean))

# Summary comparison
cat("\n--- Comparison of Unbiased vs MLE ---\n")
cat(sprintf("True sigma^2 = 225\n"))
cat(sprintf("S^2 (unbiased) = %.4f  |  bias ≈ 0\n", var_unbiased))
cat(sprintf("MLE var        = %.4f  |  bias = -sigma^2/n = %.4f\n",
            var_mle, -225 / n))

File: Chapter11/Python/Example1.py
# Chapter 11 - Σημειακή Εκτίμηση
# Example 1: Βασικοί Εκτιμητές
#   N(100, 15^2), n=50: x̄, S², σ̂²_MLE, SE(x̄)
#   Note: numpy ddof=1 -> unbiased S², ddof=0 -> MLE

import numpy as np

# Simulated sample from N(100, 15^2)
np.random.seed(42)
n           = 50
sample_data = np.random.normal(loc=100, scale=15, size=n)

# Point estimates for mean
mean_estimate = np.mean(sample_data)
print(f"Mean estimate: {mean_estimate:.4f}")

# Point estimates for variance
# Unbiased estimator (S^2) - ddof=1
var_unbiased = np.var(sample_data, ddof=1)
# MLE estimator - ddof=0
var_mle      = np.var(sample_data, ddof=0)

print(f"Unbiased variance (S^2): {var_unbiased:.4f}")
print(f"MLE variance:            {var_mle:.4f}")

# Standard deviation estimates
sd_unbiased = np.std(sample_data, ddof=1)
sd_mle      = np.std(sample_data, ddof=0)

print(f"Unbiased SD: {sd_unbiased:.4f}")
print(f"MLE SD:      {sd_mle:.4f}")

# Standard error of the mean
se_mean = np.std(sample_data, ddof=1) / np.sqrt(n)
print(f"SE of mean:  {se_mean:.4f}")

# Summary comparison
print("\n--- Comparison of Unbiased vs MLE ---")
print(f"True sigma^2 = 225")
print(f"S^2 (unbiased) = {var_unbiased:.4f}  |  bias ≈ 0")
print(f"MLE var        = {var_mle:.4f}  |  bias = -sigma^2/n = {-225/n:.4f}")
